Arm/State.lean

```diff
@@ -255,7 +255,11 @@ inductive StateField where
   | PC     : StateField
   | FLAG   : PFlag → StateField
   | ERR    : StateField
-deriving DecidableEq, Repr
+deriving DecidableEq, Repr, Hashable
+
+/-- general purpose register `x31` is used as stack pointer -/
+@[state_simp_rules]
+abbrev StateField.SP := StateField.GPR 31#5
```

Comment:
Nice! We ought to add such abbreviations for other registers as well in a future PR (and perhaps something like this is in one of the pending PRs). I know @bollu has thoughts re. notation.